src/Core/Domain/Product/SpecificPrice/ValueObject/InitialPrice.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\SpecificPrice\ValueObject;
+
+use PrestaShop\Decimal\DecimalNumber;
+
+class InitialPrice implements FixedPriceInterface
+{
+    /**
+     * Inherited from legacy.
+     * SpecificPrice->price can have certain value which means that product initial price should be used.
+     */
+    public const INITIAL_PRICE_VALUE = '-1';
+
+    public function getValue(): DecimalNumber
+    {
+        return new DecimalNumber(self::INITIAL_PRICE_VALUE);
```

Comment:
Maybe this should be stored in a class variable to avoid creating a new DecimalNumber on each getter call